pkg/operator/resource/resourceapply/core.go

```diff
@@ -3,42 +3,102 @@ package resourceapply
 import (
 	"bytes"
 	"context"
+	"crypto/md5"
 	"fmt"
+	"io"
 	"sort"
 	"strings"
 
-	"k8s.io/klog/v2"
-
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
 
 	"github.com/openshift/library-go/pkg/operator/events"
 	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
 )
 
+type CachedVersionKey struct {
+	name, kind, namespace string
+}
+
+type CachedResource struct {
+	resourceHash, resourceVersion string
```

Comment:
for future readers can you godoc these.  I think that `resourcehash` is the hash of the `required` in an ApplyFoo that is computed in case the input changes and `resourceVersion` is the received resourceVersion from the apiserver in response to an update that is comparable to the GET.

Reply:
Done